### DISCUSSION TITLE
HX711 load Cell on Wemod D1 mini

### DISCUSSION THREAD

**Discussion by antoniolespaul** [arendst/Tasmota#11116] [Q&A] [answered]
Hi, I have assembled a balance with Tasmota 9.2 on Wemos D1 mini.

The first problem is that when I calibrate the scale with the Sensor34 2 40000 command (calibration sample of 40kg), despite the calibration has been completed, at the next weigh, with the same item of calibration, it reads me about 1kg more.

The second question is, can I read the decimals of the kg?  if so how to do it?  on the command guide, I don't find this function

Thanks

Antonio

**Comment by barbudor** [comment, marked as answer]
The accuracy issue is not related to Tasmota. Tasmota just read the digital value in the interface board. If the reading is not repetitive, it may be a hardware problem (sensor, converter board, mechanical...)

My understanding of the code is that the driver is supposed to return an integer value in grams, not kg. So you should have all digits you need.
Did you used `Sensor34 4 <value>` ? 

What is the result of `Sensor34 3` (show reference weight in grams) and `Sensor34 4` (show calibrated scale value) ?

**Reply by antoniolespaul** [reply]
the web output has the unit of measure displayed in Kg, while during calibration the value of the reference weight must be entered in grams.  If I run the sensor34 3 command, I perfectly visualize the reference value I specified in grams
instead the scale learns itself 21 as a value.
Thanks

**Reply by sfromis** [reply]
When it comes to decimals, [WeightRes](https://tasmota.github.io/docs/Commands/#weightres) is probably what you're looking for.

**Reply by antoniolespaul** [reply]
> When it comes to decimals, [WeightRes](https://tasmota.github.io/docs/Commands/#weightres) is probably what you're looking for.

Thanks! It's ok!

I have some questions regarding these parameters:

This is the line command:
00:09:16 CMD: Sensor34 3
00:09:16 MQT: stat/tasmota_294006/RESULT = {"Sensor34":{"WeightRef":75000,"WeightCal":21,"WeightMax":200000,"WeightItem":0.0,"WeightChange":OFF,"WeightDelta":0}}

WeightRef = sample weight used for calibration?

WeightCal = set calibrated scale value, X10 x100 x1000... Why 21 for me?

WeightMax = 200000 g (4x50kg load Cell... It is ok?)

when i do the calibration with a weight of 75kg and it is successfully completed, and again I accurately put the same weight to measure, it returns me 77-78kg, why? there are no mechanical or other games.

Thanks

**Reply by sfromis** [reply]
AFAICT, the "WeightCal" is a conversion factor from the sensor output to weight. WeightMax is a compile-time constant in the driver.
